src/common/Text.js

```diff
@@ -60,10 +62,15 @@
                 break;
         }
         var yOffset = -(bbox.y + bbox.height / 2);
-
+        
+        if (Math.abs(context._rotation) === 90) {
+            xOffset = -(bbox.y + bbox.height / 2);
+            yOffset = 0;
+        }
```

Comment:
You're right... in its current state this offset needs some sort of trigonometric equation to determine its xOffset and yOffset... but that assumes that its parent isn't rotated? I'm not sure... maybe this implementation needs to be refactored?